src/Experiment/src/FieldBuilder.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Reflection.Metadata;
+
+namespace System.Reflection.Emit.Experimental
+{
+    // Summary:
+    //     Defines and represents a field. This class cannot be inherited.
+    public sealed class FieldBuilder : FieldInfo
+    {
+        #region Private Data Members
+        internal TypeBuilder _typeBuilder;
+        internal ModuleBuilder _moduleBuilder;
+        private string _fieldName;
+        private FieldAttributes _attributes;
+        private Type _fieldType;
+        private BlobBuilder _fieldSignature;
+        #endregion
+
+        internal EntityHandle _fieldTok;
+
+        #region Constructor
+        internal FieldBuilder(TypeBuilder typeBuilder, string fieldName, Type type,
+#pragma warning disable SA1011 // Closing square brackets should be spaced correctly
+            Type[]? requiredCustomModifiers, FieldAttributes attributes, EntityHandle fieldTok)
+#pragma warning restore SA1011 // Closing square brackets should be spaced correctly
+        {
+            ArgumentException.ThrowIfNullOrEmpty(fieldName);
```

Comment:
There are places in this review that doesn't use `ArgumentException.ThrowIfNullOrEmpty()`. I'd audit those places.

Reply:
Same for `ArgumentNullException.ThrowIfNull()`.